plugin.audio.mixcloud/default.py

```diff
@@ -767,40 +760,49 @@ def get_stream(cloudcast_key):
     global resolverid_curr
     global resolverid_orig
 
-    resolverid_start=resolverid_curr
-
-    log_if_debug('Resolverid _curr=%s _orig=%s' % (resolverid_curr,resolverid_orig))
-
-    resolvers={Resolver.auto : get_stream_local,
-               Resolver.local : get_stream_local,
-               Resolver.offliberty : get_stream_offliberty,
-               Resolver.m4a : get_stream_m4a,
-               Resolver.mixclouddownloader1 : get_stream_mixclouddownloader1,
-               Resolver.mixclouddownloader2 : get_stream_mixclouddownloader2}
-
-    strm=resolvers[resolverid_curr](cloudcast_key)
-
-    if not strm:
-        log_if_debug('Cannot solve using current resolver')
-
-        if (resolverid_orig!=Resolver.auto):
-            dialog=xbmcgui.Dialog()
-            if not dialog.yesno('MixCloud',STRLOC_COMMON_RESOLVER_ERROR):
-                return strm
-
-        while (not strm):
-            resolverid_curr=resolverid_curr+1
-            if resolverid_curr>Resolver.mixclouddownloader2:
-                resolverid_curr=Resolver.auto
-            if resolverid_curr==resolverid_start:
+    if not resolverid_curr in resolver_order:
+        resolverid_curr=resolver_order[0]
+
+    resolver_functions={Resolver.local : get_stream_local,
+                        Resolver.offliberty : get_stream_offliberty,
+                        Resolver.m4a : get_stream_m4a,
+                        Resolver.mixclouddownloader1 : get_stream_mixclouddownloader1,
+                        Resolver.mixclouddownloader2 : get_stream_mixclouddownloader2}
+
+    strm_url=None
+    strm_isexclusive=False
+    while not strm_url:
```

Comment:
check for `abortRequested` here otherwise kodi may hang on exiting

Reply:
added abortRequested() and squashed commit